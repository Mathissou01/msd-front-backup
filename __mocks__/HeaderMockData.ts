import { GetContractMenuDocument } from "../src/graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetContractMenuDocument,
      variables: {
        contractId: "1",
      },
    },
    result: {
      data: {
        contract: {
          data: {
            attributes: {
              contractMenu: {
                data: {
                  attributes: {
                    serviceLinks: [
                      {
                        name: "Carte",
                        isDisplayed: true,
                        picto: {
                          data: {
                            attributes: {
                              url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-grymas/address.svg",
                            },
                          },
                        },
                      },
                      {
                        name: "Collecte à mon adresse",
                        isDisplayed: true,
                        picto: {
                          data: null,
                        },
                      },
                      {
                        name: "Guide du tri",
                        isDisplayed: true,
                        picto: {
                          data: null,
                        },
                      },
                      {
                        name: "Faire une demande",
                        isDisplayed: true,
                        picto: {
                          data: null,
                        },
                      },
                      {
                        name: "Réduire mes déchets",
                        isDisplayed: true,
                        picto: {
                          data: null,
                        },
                        freeContentSubService: {
                          data: {
                            id: "2",
                            attributes: {
                              name: "Réduire mes déchets",
                            },
                          },
                        },
                      },
                      {
                        name: "Valoriser mes déchets",
                        isDisplayed: true,
                        picto: {
                          data: null,
                        },
                        freeContentSubService: {
                          data: {
                            id: "1",
                            attributes: {
                              name: "Valoriser mes déchets",
                            },
                          },
                        },
                      },
                    ],
                  },
                },
              },
            },
          },
        },
      },
    },
  },
];
