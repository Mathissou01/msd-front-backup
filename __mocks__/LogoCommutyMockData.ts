import { GetLogoDocument } from "../src/graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetLogoDocument,
      variables: {
        contractId: "1",
      },
    },
    result: {
      data: {
        contract: {
          data: {
            id: "1",
            attributes: {
              logo: {
                data: {
                  id: "24",
                  attributes: {
                    name: "Votre-collectivite_large.png",
                    mime: "image/png",
                    size: 21.91,
                    url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Votre-collectivite_large.png",
                    provider: "azure-storage-blob",
                    hash: "Votre_collectivite_large_40e95434cd",
                    alternativeText: "Votre-collectivite_large.png",
                  },
                },
              },
            },
          },
        },
      },
    },
  },
];
