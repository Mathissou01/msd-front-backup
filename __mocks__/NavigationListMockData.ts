import { GetContractMenuDocument } from "../src/graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetContractMenuDocument,
      variables: {
        contractId: "1",
      },
    },
    result: {
      data: {
        contract: {
          __typename: "ContractEntityResponse",
          data: {
            __typename: "ContractEntity",
            attributes: {
              __typename: "Contract",
              contractMenu: {
                __typename: "ContractMenuEntityResponse",
                data: {
                  __typename: "ContractMenuEntity",
                  attributes: {
                    __typename: "ContractMenu",
                    serviceLinks: [
                      {
                        __typename: "ComponentLinksMap",
                        name: "Carte",
                        isDisplayed: true,
                        picto: {
                          __typename: "UploadFileEntityResponse",
                          data: {
                            __typename: "UploadFileEntity",
                            attributes: {
                              __typename: "UploadFile",
                              url: "https://fake-url/address.svg",
                            },
                          },
                        },
                      },
                      {
                        __typename: "ComponentLinksCalendar",
                        name: "Collecte à mon adresse",
                        isDisplayed: true,
                        picto: {
                          __typename: "UploadFileEntityResponse",
                          data: null,
                        },
                      },
                      {
                        __typename: "ComponentLinksRecycling",
                        name: "Guide du tri",
                        isDisplayed: true,
                        picto: {
                          __typename: "UploadFileEntityResponse",
                          data: null,
                        },
                      },
                      {
                        __typename: "ComponentLinksRequest",
                        name: "Faire une demande",
                        isDisplayed: true,
                        picto: {
                          __typename: "UploadFileEntityResponse",
                          data: null,
                        },
                      },
                      {
                        __typename: "ComponentLinksFrees",
                        name: "Réduire mes déchets",
                        isDisplayed: true,
                        picto: {
                          __typename: "UploadFileEntityResponse",
                          data: null,
                        },
                        freeContentSubService: {
                          __typename: "FreeContentSubServiceEntityResponse",
                          data: {
                            __typename: "FreeContentSubServiceEntity",
                            id: "2",
                            attributes: {
                              __typename: "FreeContentSubService",
                              name: "Réduire mes déchets",
                            },
                          },
                        },
                      },
                      {
                        __typename: "ComponentLinksFrees",
                        name: "Valoriser mes déchets",
                        isDisplayed: true,
                        picto: {
                          __typename: "UploadFileEntityResponse",
                          data: null,
                        },
                        freeContentSubService: {
                          __typename: "FreeContentSubServiceEntityResponse",
                          data: {
                            __typename: "FreeContentSubServiceEntity",
                            id: "1",
                            attributes: {
                              __typename: "FreeContentSubService",
                              name: "Valoriser mes déchets",
                            },
                          },
                        },
                      },
                    ],
                  },
                },
              },
            },
          },
        },
      },
    },
  },
];
