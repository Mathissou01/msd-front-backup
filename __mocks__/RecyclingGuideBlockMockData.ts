import { GetRecyclingGuideSearchEngineDocument } from "../src/graphql/codegen/generated-types";

export const defaultMockData = {
  id: "1",
  attributes: {
    titleContent: "Un doute sur le recyclage d'un déchet",
    subtitleContent: "Trouver les consignes de tri dans ma ville",
    recyclingGuideDisplayContent: "Lampe, écran, épluchure",
    tags: {
      data: [],
    },
  },
};

export const defaultMockProviderData = [
  {
    request: {
      query: GetRecyclingGuideSearchEngineDocument,
      variables: {
        searchTerm: "cal",
        contractId: "1",
      },
    },
    result: {
      data: {
        recyclingGuideSearchEngine: [
          {
            typeName: "wasteForm",
            id: "5",
            name: "Aiguille usagee (medicale)",
          },
          {
            typeName: "wasteForm",
            id: "64",
            name: "Calculatrice",
          },
          {
            typeName: "wasteForm",
            id: "78",
            name: "Chaussons d'escalade",
          },
          {
            typeName: "wasteForm",
            id: "232",
            name: "Polystyrene expanse (calage)",
          },
          {
            typeName: "wasteForm",
            id: "271",
            name: "Stylo avec aiguille (medical)",
          },
          {
            typeName: "wasteForm",
            id: "326",
            name: "Bocal en verre",
          },
          {
            typeName: "wasteForm",
            id: "381",
            name: "Couvercle en metal (bocal)",
          },
          {
            typeName: "wasteForm",
            id: "409",
            name: "Gant et masque chirurgical ou FFP2",
          },
          {
            typeName: "wasteForm",
            id: "443",
            name: "Papier calque et millimetre ",
          },
        ],
      },
    },
  },
];
