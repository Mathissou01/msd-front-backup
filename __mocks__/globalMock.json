{
  "contract": {
    "id": "1",
    "attributes": {
      "clientName": "Grand Test",
      "clientType": "city",
      "contractStatus": "Initialisation",
      "isNonExclusive": false,
      "isRVFrance": true,
      "logo": {
        "data": {
          "id": "26",
          "attributes": {
            "name": "images.png",
            "mime": "image/png",
            "size": 2.24,
            "url": "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-root/images.png",
            "provider": "azure-storage-blob",
            "hash": "images_eb8049ec37",
            "alternativeText": null
          }
        }
      },
      "contractCustomization": {
        "data": {
          "id": "2",
          "attributes": {
            "footer": {
              "data": {
                "attributes": {
                  "accessibilityLevel": null,
                  "cguSubService": {
                    "data": {
                      "id": "1",
                      "attributes": {
                        "name": "CGU",
                        "link": null
                      }
                    }
                  },
                  "accessibilitySubService": {
                    "data": {
                      "id": "1",
                      "attributes": {
                        "name": "Accessibilité",
                        "link": null
                      }
                    }
                  },
                  "confidentialitySubService": {
                    "data": {
                      "id": "1",
                      "attributes": {
                        "name": "Confidentialité",
                        "link": null
                      }
                    }
                  },
                  "cookiesSubService": {
                    "data": {
                      "id": "1",
                      "attributes": {
                        "name": "Cookies",
                        "link": null
                      }
                    }
                  },
                  "contactUsSubService": {
                    "data": {
                      "id": "1",
                      "attributes": {
                        "name": "Contactez-nous",
                        "label": "Contactez-nous",
                        "link": null,
                        "isActivated": false
                      }
                    }
                  }
                }
              }
            }
          }
        }
      },
      "contractMenu": {
        "data": {
          "attributes": {
            "serviceLinks": [
              {
                "__typename": "ComponentLinksRecyclingGuide",
                "id": "3",
                "name": "Guide du tri",
                "isDisplayed": true,
                "picto": {
                  "data": null
                }
              },
              {
                "__typename": "ComponentLinksNews",
                "id": "2",
                "name": "Actualités",
                "isDisplayed": true,
                "picto": {
                  "data": null
                }
              },
              {
                "__typename": "ComponentLinksEvents",
                "id": "2",
                "name": "Évenements",
                "isDisplayed": true,
                "picto": {
                  "data": null
                }
              },
              {
                "__typename": "ComponentLinksFrees",
                "id": "3",
                "name": "Editorial 1",
                "isDisplayed": true,
                "freeContents": {
                  "data": [
                    {
                      "id": "1",
                      "attributes": {
                        "name": "Editorial 1"
                      }
                    }
                  ]
                },
                "picto": {
                  "data": null
                }
              },
              {
                "__typename": "ComponentLinksFrees",
                "id": "4",
                "name": "Editorial 2",
                "isDisplayed": true,
                "freeContents": {
                  "data": [
                    {
                      "id": "2",
                      "attributes": {
                        "name": "Editorial 2"
                      }
                    }
                  ]
                },
                "picto": {
                  "data": null
                }
              }
            ]
          }
        }
      }
    }
  },
  "colors": {
    "external-contrast-text": "#FFFFFF",
    "external-primary": "#0A959F",
    "calculated-primary-dark": "#006771",
    "calculated-primary-light": "#0a959f1a",
    "external-secondary": "#A3C037",
    "calculated-secondary-dark": "#719000",
    "calculated-secondary-light": "#a3c0371a",
    "external-waste-household": "#6C757D",
    "external-waste-selective": "#FFC229",
    "external-waste-bio": "#BD4602",
    "external-waste-green": "#9BCD41",
    "external-waste-glass": "#30A23D",
    "external-waste-paper": "#2068C7",
    "external-waste-other": "#FFFFFF"
  }
}
