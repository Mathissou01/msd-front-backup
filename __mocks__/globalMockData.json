{
  "contractMenu": {
    "data": {
      "attributes": {
        "serviceLinks": [
          {
            "__typename": "ComponentLinksMap",
            "name": "Carte",
            "isDisplayed": true,
            "picto": {
              "data": null
            }
          },
          {
            "__typename": "ComponentLinksCalendar",
            "name": "Collecte à mon adresse",
            "isDisplayed": true,
            "picto": {
              "data": null
            }
          },
          {
            "__typename": "ComponentLinksRequest",
            "name": "Faire une demande",
            "isDisplayed": true,
            "picto": {
              "data": null
            }
          },
          {
            "__typename": "ComponentLinksRecycling",
            "name": "Guide du tri",
            "isDisplayed": true,
            "picto": {
              "data": null
            }
          },
          {
            "__typename": "ComponentLinksFrees",
            "name": "Réduire mes déchets",
            "isDisplayed": true,
            "picto": {
              "data": null
            }
          },
          {
            "__typename": "ComponentLinksFrees",
            "name": "Valoriser mes déchets",
            "isDisplayed": true,
            "picto": {
              "data": null
            }
          }
        ]
      }
    }
  }
}
