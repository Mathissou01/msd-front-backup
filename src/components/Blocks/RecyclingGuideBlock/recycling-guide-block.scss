@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.typography" as typo;
@use "src/styles/02-tools/tools.inputs" as input;
@use "src/styles/02-tools/tools.include-media" as im;

.c-RecyclingGuideBlock {
  display: flex;
  flex-direction: column;
  gap: 32px;
  align-items: center;
  padding: 40px 16px;
  background-color: colors.$white;

  @include im.media(">=tablet") {
    padding: 60px 160px;
  }

  @include im.media(">=laptop") {
    padding: 80px 160px;
  }

  @include im.media(">=desktop") {
    padding: 80px 124px;
  }

  @include im.media(">=large-desktop") {
    padding: 80px 224px;
  }

  &__Heading,
  &__SearchContainer {
    display: flex;
    flex-direction: column;
    width: 100%;
    max-width: 648px;

    @include im.media(">=tablet") {
      max-width: 703px;
    }

    @include im.media(">=desktop") {
      max-width: 785px;
    }

    @include im.media(">=large-desktop") {
      max-width: 945px;
    }
  }

  &__Title {
    @include typo.text;

    color: colors.$expert-blue;
    font-size: 18px;
    text-align: center;
  }

  &__Subtitle {
    @include typo.heading(2);

    color: colors.$expert-blue;
    text-align: center;
  }

  &__SearchContainer {
    gap: 12px;
  }

  &__SearchBar {
    display: flex;
    flex-direction: column;
    gap: 8px;

    @include im.media(">=mobile") {
      flex-direction: row;

      button {
        width: unset;
      }
    }
  }

  &__Input {
    @include input.input;

    height: 56px;
  }

  &__Tags {
    display: flex;
    gap: 8px;
    align-items: center;
    height: 37px;
    margin-top: 12px;
    margin-right: -16px;
    overflow-x: scroll;
    overflow-y: hidden;
    -ms-overflow-style: none; /* IE and Edge */
    scrollbar-width: none; /* Firefox */
    &::-webkit-scrollbar {
      display: none;
    }
  }

  &__Tag {
    display: flex;
    flex-direction: row;
    flex-shrink: 0;
    gap: 8px;
    align-items: center;
    justify-content: center;
    height: 35px;
    padding: 7px 12px;
    color: colors.$expert-blue;
    font-size: 14px;
    line-height: 21px;
    background-color: colors.$white;
    border: 1px solid colors.$gray-200;
    border-radius: 32px;

    &::before {
      width: 16px;
      height: 16px;
      content: url("data:image/svg+xml,%3Csvg width='16' height='16' viewBox='-1 0 19 17' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M.498 8.5a7.5 7.5 0 1 0 15 0 7.5 7.5 0 0 0-15 0v0Zm2.197 5.303L13.301 3.197m0 10.606L2.695 3.197' stroke='%23767C97' stroke-width='1.5' stroke-linecap='round' stroke-linejoin='round'/%3E%3C/svg%3E");
    }
  }
}
