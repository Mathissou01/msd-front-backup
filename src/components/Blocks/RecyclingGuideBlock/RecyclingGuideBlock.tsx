import React, { useEffect, useState } from "react";
import { useRouter } from "next/router";
import { useForm } from "react-hook-form";
import {
  RecyclingGuideBlockEntity,
  SearchResult,
  useGetRecyclingGuideSearchEngineLazyQuery,
} from "../../../graphql/codegen/generated-types";
import { useContract } from "../../../hooks/useContract";
import CommonButton from "../../Common/CommonButton/CommonButton";
import "./recycling-guide-block.scss";

interface IRecyclingGuideBlock {
  data: RecyclingGuideBlockEntity;
  onUpdateRecyclingGuideSearchData: (data: (SearchResult | null)[]) => void;
}

interface IFormInput {
  searchTerm: string;
}

export default function RecyclingGuideBlock({
  data,
  onUpdateRecyclingGuideSearchData,
}: IRecyclingGuideBlock) {
  /* Static Data */
  const minLengthSearch = 3;

  /* Methods */
  function onSubmit(search: IFormInput) {
    getRecyclingGuideSearchEngine({
      variables: { contractId: contractId, searchTerm: search.searchTerm },
    });
  }

  function handleChange(event: { target: HTMLInputElement } | string) {
    if (typeof event === "string") {
      setDisabled(event.length < minLengthSearch);
      if (event.length < minLengthSearch) {
        onUpdateRecyclingGuideSearchData([]);
      }
    } else {
      setDisabled(event.target.value.length < minLengthSearch);
      if (event.target.value.length < minLengthSearch) {
        onUpdateRecyclingGuideSearchData([]);
      }
    }
  }

  /* Local Data */
  const router = useRouter();
  const titleContent = data.attributes?.titleContent;
  const subtitleContent = data.attributes?.subtitleContent;
  const recyclingGuideDisplayContent =
    data.attributes?.recyclingGuideDisplayContent;
  const tags = data.attributes?.tags?.data;
  const { register, handleSubmit } = useForm<IFormInput>();
  const [disabled, setDisabled] = useState<boolean>(true);
  const { contractId } = useContract();
  const [getRecyclingGuideSearchEngine, { data: searchData }] =
    useGetRecyclingGuideSearchEngineLazyQuery();

  useEffect(
    () => {
      if (
        searchData !== undefined &&
        searchData !== null &&
        searchData?.recyclingGuideSearchEngine
      ) {
        onUpdateRecyclingGuideSearchData([
          ...searchData.recyclingGuideSearchEngine,
        ]); // Appel de la fonction de rappel
      }
    },
    /* eslint-disable */
    [searchData],
  );

  useEffect(() => {
    if (router.query) {
      const searchData: IFormInput = { searchTerm: `${router.query.search}` };
      onSubmit(searchData);
      handleChange(`${router.query.search}`); // Manually call handleChange
    }
  }, [router.query.search]);

  return (
    <section className="c-RecyclingGuideBlock">
      <div className="c-RecyclingGuideBlock__Heading">
        <span className="c-RecyclingGuideBlock__Title">{titleContent}</span>
        <h2
          id="findRecyclingGuideSubtitle"
          className="c-RecyclingGuideBlock__Subtitle"
        >
          {subtitleContent}
        </h2>
      </div>
      <div className="c-RecyclingGuideBlock__SearchContainer">
        <form
          className="c-RecyclingGuideBlock__SearchBar"
          onSubmit={handleSubmit(onSubmit)}
        >
          <input
            className="c-RecyclingGuideBlock__Input"
            placeholder={recyclingGuideDisplayContent}
            {...register("searchTerm", {
              required: true,
              minLength: minLengthSearch,
            })}
            type="text"
            defaultValue={router.query.search}
            onChange={handleChange}
            title="Saisir votre recherche"
            aria-labelledby="findRecyclingGuideSubtitle"
          />
          <CommonButton
            label="Rechercher"
            picto="search"
            type="submit"
            style="primary"
            fontStyle="fontLarge"
            paddingStyle="paddingLarge"
            isFullWidth={true}
            isDisabled={disabled}
          />
        </form>
        {tags && (
          <div className="c-RecyclingGuideBlock__Tags">
            {tags.map((tag, index) => {
              if (tag.attributes?.name) {
                return (
                  <button
                    key={index}
                    className="c-RecyclingGuideBlock__Tag"
                    type="button"
                  >
                    {tag.attributes.name}
                  </button>
                );
              }
            })}
          </div>
        )}
      </div>
    </section>
  );
}
