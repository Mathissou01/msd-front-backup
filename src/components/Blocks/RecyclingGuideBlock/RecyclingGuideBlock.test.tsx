import { render, screen } from "@testing-library/react";
import { defaultMockData } from "../../../../__mocks__/RecyclingGuideBlockMockData";
import RecyclingGuideBlock from "./RecyclingGuideBlock";

describe("RecyclingGuideBlock", () => {
  it("renders", () => {
    const { container } = render(
      <RecyclingGuideBlock data={defaultMockData.data} />,
    );
    const title = screen.getByText("Un doute sur le recyclage d'un d√©chet");
    expect(title).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
