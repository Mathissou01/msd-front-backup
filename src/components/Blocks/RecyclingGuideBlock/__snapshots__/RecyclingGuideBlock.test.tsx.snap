// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders one tip 1`] = `
<div>
  <section
    class="c-RecyclingGuideBlock"
  >
    <div
      class="c-RecyclingGuideBlock__Heading"
    >
      <span
        class="c-RecyclingGuideBlock__Title"
      >
        Un doute sur le recyclage d'un déchet
      </span>
      <h2
        class="c-RecyclingGuideBlock__Subtitle"
      >
        Trouver les consignes de tri dans ma ville
      </h2>
    </div>
    <div
      class="c-RecyclingGuideBlock__SearchZone"
    >
      <div
        class="c-RecyclingGuideBlock__SearchBar"
      >
        <input
          class="c-RecyclingGuideBlock__Input"
          placeholder="Lampe, écran, épluchure"
          type="text"
        />
        <button
          class="c-CommonButton c-CommonButton_primary c-CommonButton_paddingLarge c-CommonButton_fontLarge c-CommonButton_fullWidth"
          type="button"
        >
          <div
            class="c-CommonButton__Picto c-CommonButton__Picto_search"
          />
          Rechercher
        </button>
      </div>
      <div
        class="c-RecyclingGuideBlock__Tags"
      >
        <button
          class="c-RecyclingGuideBlock__Tag"
          type="button"
        >
          Bouteille plastique
        </button>
        <button
          class="c-RecyclingGuideBlock__Tag"
          type="button"
        >
          Carton de pizza
        </button>
        <button
          class="c-RecyclingGuideBlock__Tag"
          type="button"
        >
          Papier collant
        </button>
      </div>
    </div>
  </section>
</div>
`;
